### ISSUE TITLE
Page Load Detection

### ISSUE THREAD

**Issue by laserlemon** [teampoltergeist/poltergeist#434]
Poltergeist itself has the ability to track what it considers to be new page loads within the JavaScript browser by incrementing the `@page_id` value. I think it could be a valuable feature to communicate that a new page has been instantiated back to the socket. This could pave the way for push notification of page load rather than polling for a URL change (which doesn't always change).

In my first try to communicate the incremented `@page_id` back to the Ruby browser, lines of communication kept crossing and `find` commands were returning the incremented `@page_id` response rather than the response resulting from the `find` command.

Long story short, I haven't cracked that nut yet but I'd like to know if it's a feature that would be considered for inclusion. Thanks!
